src/area.rs

```diff
@@ -0,0 +1,157 @@
+use bevy::prelude::*;
+use bevy::sprite::collide_aabb::collide;
+
+use crate::player::Player;
+
+pub struct AreaPlugin;
+
+const AREA_COUNT: usize = 2;
+
+struct GameAreas {
+    areas: [Area; AREA_COUNT],
+}
+
+impl Default for GameAreas {
+    fn default() -> Self {
+        let passage_east = Passage {
+            transform: Transform::from_xyz(1280. / 2. - 15., 0., 1.),
```

Comment:
Bunch of magic numbers here for now, lots of space for improvement in giving them names and appropriate places to live. Don't think we need to rush for perfection immediately though, their proper places will likely arise from other feature implementations.